### PR TITLE
Frozen - Symbolize optimus

### PR DESCRIPTION
**Pull request by T-Dnzt**
### Frozen 
This PR is not complete and still requires work. To prioritize important tasks, the work on this PR has to be paused.
See [Trello Card](https://trello.com/c/Atr6Lthe/152-enforce-symbols-in-optimus-prime-bumblebee)

What has been done
- Symbolize the sources, transformers and destinations
- Update specs for the sources, transformers and destinations

What's left
- Finish converting the specs and testing the ETL classes.

---

This PR enforces the use of symbols in Optimus Prime. To do this, it will convert the keys of each streamed message that is a hash.

All spec hash inputs have also been changed to use symbols.

### CODE REVIEW

File: lib/optimus_prime/transformers/cast_string.rb [hunk 1]
```diff
@@ -24,10 +24,9 @@ module OptimusPrime
   module Transformers
     class CastString < Destination
       # type_map - Hash specifying the types like in the example above
-      # stringify - Optional Boolean you can set to false to
       #             NOT convert Symbol keys of the type_map to Strings
```
Comment (simonbnrd) [inline lib/optimus_prime/transformers/cast_string.rb]: You should remove this line. We don't need it anymore.

File: spec/optimus_prime/transformers/record_filter_spec.rb [hunk 2]
```diff
@@ -16,27 +16,27 @@
```
Comment (simonbnrd) [inline spec/optimus_prime/transformers/record_filter_spec.rb]: context 'stringify' to remove/change

**Comment by simonbnrd** [comment]
:+1:

**Comment by omarkhan** [comment]
Bigquery doesn't seem to like this

**Comment by T-Dnzt** [comment]
@omarkhan Yes, currently working on it, coming soon :tm:

**Pull request closed without merging**